File: JAVA/ArrayList_210902/src/codes/datas/StudentData.java
package codes.datas;

public class StudentData {
	String name;
	int birthYear;
	
	public StudentData(String name, int birthYear) {
		super();
		this.name = name;
		this.birthYear = birthYear;
	}
	
	@Override
	public String toString() {
		String message = String.format("%s(%d년생)", this.name, this.birthYear);
		
		return message;
	}

	
	@Override
	public boolean equals(Object obj) {
		// 일단 형태가 맞는지부터
		if(!(obj instanceof StudentData)) return false;
		
		if(otherStd.name == this.name && otherStd.birthYear == this.birthYear) return true;
		else return false;
	}
}

File: ListView_210820/app/src/main/java/com/gukjang/listview_210820/datas/StudentData.kt
package com.gukjang.listview_210820.datas

class StudentData(val name : String,                // 생성자
                  val birthYear : Int,
                  val  address : String) {

// 학생의 기능으로 -> 나이를 계산해서 알려주는 기능을 함수로
    fun getKoreanAgeByYear(year : Int) : Int {
        val age = year - this.birthYear + 1

        return age
    }
}